ctrd/container.go

```diff
@@ -137,50 +132,68 @@ func (c *Client) execContainer(ctx context.Context, process *Process) error {
 		return errors.Wrap(err, "failed to exec process")
 	}
 
-	errCh := make(chan error, 1)
-	defer close(errCh)
-
-	go func() {
-		var msg *Message
+	var msg *Message
+	defer func() {
+		if msg != nil {
```

Comment:
msg won’t be nil here. or use fast return `if msg == nil return` to reduce nested level.